### PR TITLE
fix: Nuxt module detection not checking using actual Nuxt instance

### PR DESCRIPTION
**Pull request by BobbieGoede**
## Types of changes

- [ ] New Detector
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
From what I can tell most of the Nuxt module detectors probably don't work on recent Nuxt versions (maybe they don't work since Nuxt v3?). I was mostly interested in fixing the detector for `@nuxtjs/i18n` but I quickly ran through the other modules as well (I haven't tested those).

## Checklist:

- [ ] I have updated the README accordingly 
- [x] I have added the icon of my detector in `icons/`

### CODE REVIEW

File: detectors/nuxt.modules.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
       "url": "https://http.nuxtjs.org"
     },
     "detectors": {
-      "js": "window.$nuxt?.$options?.context?.app?.$http"
+      "js": "window.$nuxt?.$options?.context?.app?.$http || window.__unctx__?.get('nuxt-app')?.use()?.$http"
```
Comment (danielroe) [inline detectors/nuxt.modules.json:21]: we probably can omit for the Nuxt 2-only modules.

Comment (BobbieGoede) [reply detectors/nuxt.modules.json]: Agreed, I tried to check and update which modules have Nuxt 3 support and if the detection would still work by checking for the property.

File: detectors/nuxt.modules.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
       "url": "https://github.com/nuxt-community/apollo-module"
```
Comment (danielroe) [inline detectors/nuxt.modules.json:40]:
```suggestion
      "url": "https://github.com/nuxt-modules/apollo"
```

File: detectors/nuxt.modules.json [hunk 3]
```diff
@@ -84,7 +84,7 @@
       "url": "https://github.com/nuxt-community/device-module"
```
Comment (danielroe) [inline detectors/nuxt.modules.json:84]:
```suggestion
      "url": "https://github.com/nuxt-modules/device"
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**